src/java.management/share/classes/javax/management/modelmbean/ModelMBeanAttributeInfo.java

```diff
@@ -116,63 +116,15 @@
  * @since 1.5
  */
 
-@SuppressWarnings("serial")  // serialVersionUID is not constant
 public class ModelMBeanAttributeInfo
     extends MBeanAttributeInfo
     implements DescriptorAccess {
 
-    // Serialization compatibility stuff:
-    // Two serial forms are supported in this class. The selected form depends
-    // on system property "jmx.serial.form":
-    //  - "1.0" for JMX 1.0
-    //  - any other value for JMX 1.1 and higher
-    //
-    // Serial version for old serial form
-    private static final long oldSerialVersionUID = 7098036920755973145L;
-    //
-    // Serial version for new serial form
-    private static final long newSerialVersionUID = 6181543027787327345L;
-    //
-    // Serializable fields in old serial form
-    private static final ObjectStreamField[] oldSerialPersistentFields =
-    {
-      new ObjectStreamField("attrDescriptor", Descriptor.class),
-      new ObjectStreamField("currClass", String.class)
-    };
-    //
-    // Serializable fields in new serial form
-    private static final ObjectStreamField[] newSerialPersistentFields =
-    {
-      new ObjectStreamField("attrDescriptor", Descriptor.class)
-    };
-    //
-    // Actual serial version and serial form
-    private static final long serialVersionUID;
-    /**
-     * @serialField attrDescriptor Descriptor The {@link Descriptor}
-     * containing the metadata corresponding to this attribute
-     */
-    private static final ObjectStreamField[] serialPersistentFields;
-    private static boolean compat = false;
-    static {
-        try {
-            GetPropertyAction act = new GetPropertyAction("jmx.serial.form");
-            @SuppressWarnings("removal")
-            String form = AccessController.doPrivileged(act);
-            compat = (form != null && form.equals("1.0"));
-        } catch (Exception e) {
-            // OK: No compat with 1.0
-        }
-        if (compat) {
-            serialPersistentFields = oldSerialPersistentFields;
-            serialVersionUID = oldSerialVersionUID;
-        } else {
-            serialPersistentFields = newSerialPersistentFields;
-            serialVersionUID = newSerialVersionUID;
-        }
-    }
-    //
-    // END Serialization compatibility stuff
+        private static final long serialVersionUID = 6181543027787327345L;
+        private static final ObjectStreamField[] serialPersistentFields =
```

Comment:
API doc coment for serial persistent fields should have been preserved.
```suggestion
        /**
         * @serialField attrDescriptor Descriptor The {@link Descriptor}
         * containing the metadata corresponding to this attribute
         */
        private static final ObjectStreamField[] serialPersistentFields =
```